Lib/test/test_fcntl.py

```diff
@@ -138,6 +136,39 @@ def test_flock(self):
         self.assertRaises(ValueError, fcntl.flock, -1, fcntl.LOCK_SH)
         self.assertRaises(TypeError, fcntl.flock, 'spam', fcntl.LOCK_SH)
 
+    def test_lockf(self):
+
+        self.f = open(TESTFN, 'wb+')
+
+        self.assertRaises(TypeError, fcntl.lockf, self.f, "foo")
+        self.assertRaises(TypeError, fcntl.lockf, self.f, fcntl.LOCK_UN, "foo")
+        self.assertRaises(ValueError, fcntl.lockf, self.f, -256)
+        self.assertRaises(ValueError, fcntl.lockf, self.f, 256)
+
+        fcntl.lockf(self.f, fcntl.LOCK_EX | fcntl.LOCK_NB)
+
+        pid = os.fork()
+        if pid == 0:
+            rval = 2
+            try:
+                fcntl.lockf(open(self.f.name, self.f.mode), fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
This line causes a `ResourceWarning` since the open file handle is not closed. Perhaps assign it to a variable and make sure it's closed in the test.

```
➜  cpython git:(master) ✗ ./python.exe -m unittest -v test.test_fcntl.TestFcntl.test_lockf
struct.pack:  b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00'
test_lockf (test.test_fcntl.TestFcntl) ... /Users/karthikeyansingaravelan/stuff/python/cpython/Lib/test/test_fcntl.py:154: ResourceWarning: unclosed file <_io.BufferedRandom name='@test_27262_tmp'>
  fcntl.lockf(open(self.f.name, self.f.mode), fcntl.LOCK_EX | fcntl.LOCK_NB)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
ok

----------------------------------------------------------------------
Ran 1 test in 0.008s

OK
```